Tools/AP_Periph/AP_Periph.cpp

```diff
@@ -416,15 +416,29 @@ void AP_Periph_FW::update()
     node_stats.update();
 #endif
 
-    static uint32_t last_led_ms;
     uint32_t now = AP_HAL::millis();
-    if (now - last_led_ms > 1000) {
-        last_led_ms = now;
+
 #ifdef HAL_GPIO_PIN_LED
+    static uint32_t last_led_ms;
+    static uint32_t next_led_ms = 1000;
+    if (now - last_led_ms > next_led_ms) {
+        last_led_ms = now;
         if (!no_iface_finished_dna) {
```

Comment:
Doesn't this mean the code will be non-functional, depending on whether it has dna'd or not?